docs/source/bottleneck.rst

```diff
@@ -17,15 +17,38 @@ where [args] are any number of arguments to `script.py`, or run
 ``python -m torch.utils.bottleneck -h`` for more usage instructions.
 
 .. warning::
-    Because your script will be profiled, please ensure that it exits in a 
+    Because your script will be profiled, please ensure that it exits in a
     finite amount of time.
 
 .. warning::
     Due to the asynchronous nature of CUDA kernels, when running against
     CUDA code, the cProfile output and CPU-mode autograd profilers may
-    not show correct timings. In this case, the CUDA-mode autograd
-    profiler is better at assigning blame to the relevant operator(s).
+    not show correct timings: the reported CPU time reports the amount of time
+    used to launch the kernels but does not include the time the kernel
+    spent executing on a GPU unless the operation does a synchronize.
+    In this case, the CUDA-mode autograd profiler may be helpful.
+
+.. warning::
+    To decide which (CPU-only-mode or CUDA-mode) autograd profiler output to
+    look at, you should first check if your script is CPU-bound
+    ("CPU total time is much greater than CUDA total time").
+    If it is CPU-bound, looking at the results of the CPU-mode autograd
+    profiler will help. If on the other hand your script spends most of its
+    time executing on the GPU, then it makes sense to start
+    looking for responsible CUDA operators in the output of the CUDA-mode
+    autograd profiler.
+
+    Of course the reality is much more complicated and your script might not be
+    in one of those two extremes depending on the part of the model you're
+    evaluating. If the profiler outputs don't help, you could try looking at
+    the result of :func:`torch.autograd.profiler.emit_nvtx()` with ``nvprof``.
```

Comment:
Extra disclaimer: NVTX overhead is really high and often gives a pretty heavily skewed timeline.